Create a simple README (#7)
NationalLibraryOfNorway/warehouse-logistics-service: README.md

# Warehouse Logistics Service

Hermes WLS (Warehouse and Logistics Service) functions as a middleware between NLNs (National Library of Norway) catalogues and storage systems.

> The goal with the service is to unite the different storage systems and magazines under the same service.

Reviewer: ```suggestion
The goal with the service is to unite the different storage systems and catalogues with a common interface.
This should help us in decoupling the storage systems and catalogues from each other and make it easier to introduce new ones in the future.
```

## Running the application

> Hermes WLS runs on Java 21.

Reviewer: Too much information, need to condense it even more 😉

Author: I could remove the "21"

### Building

> If using CLI exclusively, you can build and run with the following commands:

Reviewer: ```suggestion
To build and run this application use Maven and the following commands:
```

```shell
# Build the jar with the maven-jar-plugin
mvn jar:jar

# Run it locally
cd target/
java -jar wls.jar
```

> Otherwise we recommend using [IntelliJ IDEA](https://www.jetbrains.com/idea/) to both build and deploy locally.
> The default Spring run configuration for the application works well.

Reviewer: ````suggestion
It's possible to build a Docker image from source using the following commands:
```
... commands
```
A pre-built image can be found on NLNs local Harbor instance under the `mlt` namespace.
 
For local development and testing an IDE like [IntelliJ IDEA](https://www.jetbrains.com/idea/) is recommended.
Its default Spring run configuration for the application works well.
To test the service with authentication set the "Active profiles" variable to `stage`, and provide the `KEYCLOAK_ISSUER_URI` variable (see below).
Keycloak in stage is set up with a test client.
````

### Configuration

The following environment variables are relevant to configuring the application:

> - `KEYCLOAK_ISSUER_URI`: Is used to point at the Keycloak server used for authentication
>   - required for local testing, and when using it in staging
> - `KEYCLOAK_ISSUER_URI_PROD`: Is used to point at the Keycloak server used for authentication
>   - required when deployed to k8s and used in production environments, usually with the "prod" profile

Reviewer: `KEYCLOAK_ISSUER_URI_PROD` does not exist and isn't used in the code.

```suggestion
- `KEYCLOAK_ISSUER_URI`: Points to the Keycloak resource server used for authentication and authorzation
  - Make sure to use appropriate URI in stage and production environment
```

Author: Was confused by the k8s configuration, this is indeed the case

### Deploying

> Deploying is as simple as running the jar.
> When running locally, make sure to activate the "stage" profile with Spring.

Reviewer: ```suggestion
When deploying locally, make sure to activate the "stage" profile with Spring, and provide the KEYCLOAK_ISSUER_URI.
Easiest way to test the functionality is to use the provided Swagger UI at http://localhost:8080/swagger
```

```shell
java -DKEYCLOAK_ISSUER_URI=<url> -Dspring.profiles.active=stage -jar wls.jar
```

> 

Reviewer: Mention pipeline deployment
